File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeStartOkJobIndexMessage.scala





















// Generated on 05/03/2013 10:34:56
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeStartOkJobIndexMessage extends MessageDeserializer {
  val messageId = 5819

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeStartOkJobIndexMessage.messageId

    val jobs: Seq[Int]

    def serialize(buf: Buffer) {
      buf.writeUShort(jobs.length)
      for (entry <- jobs) {
        buf.writeInt(entry)
      }
    }

  }

  protected def create(jobs: Seq[Int]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val jobs = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }

    create(jobs)
  }

}

object ExchangeStartOkJobIndexMessage extends ExchangeStartOkJobIndexMessage {
  case class MessageImpl(jobs: Seq[Int]) extends Message
  type Target = MessageImpl

  override def create(jobs: Seq[Int]) = MessageImpl(jobs)

  def apply(jobs: Seq[Int]) = create(jobs)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeStartOkJobIndexMessage.java


// Generated on 05/08/2013 19:37:55
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeStartOkJobIndexMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5819;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int[] jobs;
    
    public ExchangeStartOkJobIndexMessage() { }
    
    public ExchangeStartOkJobIndexMessage(int[] jobs) {
        this.jobs = jobs;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(jobs.length);
        for (int entry : jobs) {
            buf.writeInt(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        jobs = new int[limit];
        for (int i = 0; i < limit; i++) {
            jobs[i] = buf.readInt();
        }
    }
    
}
